types/process.luau

```diff
@@ -5,34 +5,127 @@ export type SpawnOptionsStdioKind = "default" | "inherit" | "forward" | "none"
 
-	A dictionary of options for `process.spawn`, with the following available values:
+	A dictionary of options for `process.create`, with the following available values:
 
 	* `cwd` - The current working directory for the process
 	* `env` - Extra environment variables to give to the process
 	* `shell` - Whether to run in a shell or not - set to `true` to run using the default shell, or a string to run using a specific shell
 	* `stdio` - How to treat output and error streams from the child process - see `SpawnOptionsStdioKind` and `SpawnOptionsStdio` for more info
-	* `stdin` - Optional standard input to pass to spawned child process
 ]=]
 export type SpawnOptions = {
 	cwd: string?,
 	env: { [string]: string }?,
 	shell: (boolean | string)?,
+}
+
+--[=[
+	@interface ExecuteOptions
+	@within Process
+
+	A dictionary of options for `process.exec`, with the following available values:
+
+	* `cwd` - The current working directory for the process
+	* `env` - Extra environment variables to give to the process
+	* `shell` - Whether to run in a shell or not - set to `true` to run using the default shell, or a string to run using a specific shell
+	* `stdio` - How to treat output and error streams from the child process - see `SpawnOptionsStdioKind` and `ExecuteOptionsStdio` for more info
+	* `stdin` - Optional standard input to pass to executed child process
+]=]
+export type ExecuteOptions = SpawnOptions & {
 	stdio: (SpawnOptionsStdioKind | SpawnOptionsStdio)?,
 	stdin: string?, -- TODO: Remove this since it is now available in stdio above, breaking change
 }
 
+--[=[
+	@class ChildProcessReader
+	@within Process
+
+	A reader class to read data from a child process' streams in realtime.
+]=]
+local ChildProcessReader = {}
+
+--[=[
+	@within ChildProcessReader
+
+	Reads a chunk of data (specified length or a default of 8 bytes at a time) from 
+	the reader as a string. Returns nil if there is no more data to read.
+
+	This function may yield until there is new data to read from reader, if all data
+	till present has already been read, and the process has not exited.
+
+	@return The string containing the data read from the reader
+]=]
+function ChildProcessReader:read(chunkSize: number?): string?
+	return nil :: any
+end
+
+--[=[
+	@within ChildProcessReader
+
+	Reads all the data currently present in the reader as a string.
+	This function will yield until the process exits.
+
+	@return The string containing the data read from the reader
+]=]
+function ChildProcessReader:readToEnd(): string
+	return nil :: any
+end
+
+--[=[
+	@class ChildProcessWriter
+	@within Process
+
+	A writer class to write data to a child process' streams in realtime.
+]=]
+local ChildProcessWriter = {}
+
+--[=[
+	@within ChildProcessWriter
+
+	Writes a buffer or string of data to the writer.
+
+	@param data The data to write to the writer
+]=]
+function ChildProcessWriter:write(data: buffer | string): ()
+	return nil :: any
+end
+
 --[=[
 	@interface SpawnResult
```

Comment:
I think this is meant to be `ChildProcess` instead of `SpawnResult`.

*(also, sorry - I suck at this GitHub commenting thing)*

Reply:
Could rename this, yeah - `process.create` was formerly `process.spawn` hence `SpawnResult`.

Reply:
Resolved in https://github.com/lune-org/lune/pull/211/commits/6a713a881f49c136ecd8e30493dcc7526e5abc1d.